src/app/controllers/OverviewController.ts

```diff
@@ -373,29 +364,72 @@ export default class OverviewController {
+			}
+
+			const nodeDeposit = new BigNumber(cur.rocketpool.node_deposit_balance.toString()).dividedBy(new BigNumber(1e9))
+			const rewards = new BigNumber(fieldVal.toString()).minus(VALIDATOR_32ETH)
+
+			const nodeShare = nodeDeposit.dividedBy(VALIDATOR_32ETH).toNumber()
+			const rewardNode = new BigNumber(rewards).multipliedBy(new BigNumber(nodeShare))
+			const rewardNodeFromPool = new BigNumber(rewards)
+				.multipliedBy(new BigNumber(1 - nodeShare))
+				.multipliedBy(new BigNumber(cur.rocketpool.minipool_node_fee.toString()))
+
+			// if negative, rocketpool reimburces the rETH holder by taking it from the node operator
+			let wholeBalance
+			if (rewards.isNegative()) {
+				wholeBalance = nodeDeposit.minus(rewards)
+				// if less than what the user deposited upfront, he lost everything and we put the balance to 0
+				// the validator can leak below that of course, but that's now at the expense of the rETH holders
+				if (wholeBalance.isNegative()) {
+					wholeBalance = 0
+				}
+			} else {
+				wholeBalance = nodeDeposit.plus(rewardNode).plus(rewardNodeFromPool)
+			}
 
-			const rewards = new BigNumber(fieldVal.toString()).minus(cur.data.effectivebalance)
-			const nodeOperatorRewards = new BigNumber(rewards)
-				.multipliedBy(new BigNumber('1').plus(new BigNumber(cur.rocketpool.minipool_node_fee.toString())))
-				.dividedBy('2')
-			const wholeBalance = new BigNumber('16000000000').plus(nodeOperatorRewards)
 			return wholeBalance.multipliedBy(new BigNumber(cur.share == null ? 1 : cur.share))
 		})
 	}
 
-	private sumBigIntPerformanceRpl(validators: Validator[], field: (cur: Validator) => BigNumber): BigNumber {
+	private sumBigIntPerformanceRP(validators: Validator[], field: (cur: Validator) => BigNumber): BigNumber {
 		return sumBigInt(validators, (cur) => {
 			if (!cur.rocketpool) return field(cur)
-			if (!cur.rocketpool.node_address) return field(cur)
+			if (!cur.rocketpool.node_address || !cur.rocketpool.node_deposit_balance) return field(cur)
+
+			let fieldResolved = field(cur)
+			if (fieldResolved.s == null && fieldResolved.e == null) {
+				fieldResolved = new BigNumber(0)
+			}
+
+			const nodeDeposit = new BigNumber(cur.rocketpool.node_deposit_balance.toString()).dividedBy(new BigNumber(1e9))
+			const rewards = new BigNumber(fieldResolved.toString())
+
+			const nodeShare = nodeDeposit.dividedBy(VALIDATOR_32ETH).toNumber()
+
+			const rewardNode = new BigNumber(rewards).multipliedBy(new BigNumber(nodeShare))
+			const rewardNodeFromPool = new BigNumber(rewards)
+				.multipliedBy(new BigNumber(1 - nodeShare))
+				.multipliedBy(new BigNumber(cur.rocketpool.minipool_node_fee.toString()))
+
+			// if negative, rocketpool reimburces the rETH holder by taking it from the node operator
+			let resultReward
+			if (rewards.isNegative()) {
+				resultReward = rewards
+				if (resultReward.isNegative()) {
+					resultReward = 0
+				}
```

Comment:
This is weird, `resultReward` is always negative whenever `rewards` is negative too. You should be able to shorten this piece of code like this:
```suggestion
			if (rewards.isNegative()) {
				resultReward = 0
```

Reply:
You are right, that one's weird. I fixed the second if, I meant to check if the user lost their entire deposit balance, keep it at that.